packages/editor-ui/src/components/TagsDropdown.vue

```diff
@@ -1,236 +1,191 @@
-				}
-			},
-		);
-
-		onMounted(() => {
-			const select = selectRef.value?.$refs?.innerSelect as
-				| { $refs: { input: Element } }
-				| undefined;
-			if (select) {
-				const input = select.$refs.input as Element | undefined;
-				if (input) {
-					input.setAttribute('maxlength', `${MAX_TAG_NAME_LENGTH}`);
-					input.addEventListener('keydown', (e: Event) => {
-						const keyboardEvent = e as KeyboardEvent;
-						// events don't bubble outside of select, so need to hook onto input
-						if (keyboardEvent.key === 'Escape') {
-							emit('esc');
-						} else if (keyboardEvent.key === 'Enter' && filter.value.length === 0) {
-							preventUpdate.value = true;
-
-							emit('blur');
-
-							if (typeof selectRef.value?.blur === 'function') {
-								selectRef.value.blur();
-							}
-						}
-					});
-				}
-			}
+const selectRef = ref<InstanceType<typeof N8nSelect>>();
+const tagRefs = ref<Array<InstanceType<typeof N8nOption>>>();
+const createRef = ref<InstanceType<typeof N8nOption>>();
 
-			props.eventBus?.on('focus', onBusFocus);
+const filter = ref('');
+const focused = ref(false);
+const preventUpdate = ref(false);
 
-			void tagsStore.fetchAll();
-		});
+const container = ref<HTMLDivElement | undefined>();
 
-		onBeforeUnmount(() => {
-			props.eventBus?.off('focus', onBusFocus);
-		});
+const dropdownId = uuid();
 
-		function onBusFocus() {
-			focusOnInput();
-			focusFirstOption();
-		}
+const options = computed<ITag[]>(() => {
+	return props.allTags.filter((tag: ITag) => tag && tag.name.includes(filter.value));
+});
 
-		function filterOptions(value = '') {
-			filter.value = value;
-			void nextTick(() => focusFirstOption());
-		}
+const appliedTags = computed<string[]>(() => {
+	return props.modelValue.filter((id: string) => props.tagsById[id]);
+});
 
-		async function onCreate() {
-			const name = filter.value;
-			try {
-				const newTag = await tagsStore.create(name);
-				emit('update:modelValue', [...props.modelValue, newTag.id]);
-
-				filter.value = '';
-			} catch (error) {
-				showError(
-					error,
-					i18n.baseText('tagsDropdown.showError.title'),
-					i18n.baseText('tagsDropdown.showError.message', { interpolate: { name } }),
-				);
-			}
+watch(
+	() => props.allTags,
+	() => {
+		if (props.modelValue.length !== appliedTags.value.length) {
+			emit('update:modelValue', appliedTags.value);
 		}
-
-		function onTagsUpdated(selected: string[]) {
-			const manage = selected.find((value) => value === MANAGE_KEY);
-			const create = selected.find((value) => value === CREATE_KEY);
-
-			if (manage) {
-				filter.value = '';
-				uiStore.openModal(TAGS_MANAGER_MODAL_KEY);
-				emit('blur');
-			} else if (create) {
-				void onCreate();
-			} else {
-				setTimeout(() => {
-					if (!preventUpdate.value) {
-						emit('update:modelValue', selected);
+	},
+);
+
+onMounted(() => {
+	const select = selectRef.value?.$refs?.innerSelect as { $refs: { input: Element } } | undefined;
```

Comment:
Uff :D A better approach might be to expose the `innerSelect` in `packages/design-system/src/components/N8nSelect/Select.vue` so that it's accessible without accessing it through $refs.

So:
```ts
// packages/design-system/src/components/N8nSelect/Select.vue

....
defineExpose({
	focus,
	blur,
	focusOnInput,
	innerSelect,
});
```

Then you could just do:

```
const select = selectRef.value?.innerSelect;

	if (select) {
		const input = select.input;
```

Reply:
Got it, thanks!